static/js/notifications.js

```diff
@@ -402,6 +402,17 @@ function message_is_notifiable(message) {
         return false;
     }
 
+    // The notification should only be displayed once. To prevent multiple
+    // tabs from opening a notification each for their own, we store the
+    // notification ID in localStorage and check if a notification was
+    // already opened.
+    if (typeof(Storage) !== "undefined") {
```

Comment:
There is a method in `static/js/localstorage.js` called `supports_localstorage` which is a bit more extensive?

I am not sure if that is required here, but you might want to take a look.